### ISSUE TITLE
[feature] @OA\RequestBody, but for "query"

### ISSUE THREAD

**Issue by arjanfrans** [nelmio/NelmioApiDocBundle#1721]
**Version** 
I am using  version `4.0@dev`.

**Problem**
I define all my request parameters in a "Model" class. By using the `@OA\RequestBody` annotation I get my parameters documented automatically. This works great for all POST/PATCH/PUT requests that have a body. For GET requests I also like to define my parameters in a Model class.

As a "workaround" to document the endpoints I put "query" in the description. However, I can then not use the "Try it out" feature. Here is my implementation:

![image](https://user-images.githubusercontent.com/3588470/94018362-d2260580-fdb0-11ea-9af9-2f53f11511d5.png)
![image](https://user-images.githubusercontent.com/3588470/94022109-0bf90b00-fdb5-11ea-8ee3-f8e202eacd87.png)

**Feature**
What I'd like to see is an `@OA\RequestBody` annotation for query parameters (`@OA\RequestQuery`?) so that I can define my query parameters in my Model class.

My Model classes contains the types and validations, one place where everything is defined. I don't want to define `OA\Parameter` annotations on my route too.

**Comment by GuilhemN** [comment]
This is actually supported using openapi 3 features, please see https://github.com/nelmio/NelmioApiDocBundle/issues/1596#issuecomment-650008083.

In summary, you should use an usual `@Parameter` with `@Model` and the `explode` field.

**Comment by GuilhemN** [comment]
Closing as fixed, feel free to reopen if this is not the case.

**Comment by Luehrsen** [comment]
Does anybody have a working implementation for this? 

Currently doing 
```
     * @OA\Parameter(
     *   name="query",
     *   in="query",
     *   style="form",
     *   explode=false,
     *   @OA\Schema(
     *     ref=@Model(type=CharacterFilter::class, groups={"read"})
     *   )
     * )
```

and getting 

<img width="453" alt="Bildschirmfoto 2022-01-06 um 21 29 08" src="https://user-images.githubusercontent.com/1105871/148447355-274e8ef5-e79c-4d5d-ac91-4002eb2bea18.png">

Side question: Does the bundle pick up parameters that are in traits?

**Comment by PetrNovotnyGS** [comment]
> Does anybody have a working implementation for this?
> 
> Currently doing
> 
> ```
>      * @OA\Parameter(
>      *   name="query",
>      *   in="query",
>      *   style="form",
>      *   explode=false,
>      *   @OA\Schema(
>      *     ref=@Model(type=CharacterFilter::class, groups={"read"})
>      *   )
>      * )
> ```
> 
> and getting
> 
> <img alt="Bildschirmfoto 2022-01-06 um 21 29 08" width="453" src="https://user-images.githubusercontent.com/1105871/148447355-274e8ef5-e79c-4d5d-ac91-4002eb2bea18.png">
> 
> Side question: Does the bundle pick up parameters that are in traits? Edit: Anwser, yes! But those seem to be cached, so you have to clear caches to make them visible in the doc.

Unfortunatelly not:(

Currently using this workaround
![image](https://user-images.githubusercontent.com/11830741/167984770-f58ccedf-b0de-470f-9989-3bc157e755e7.png)

But it would be perfect, if 
new OAT\Parameter(name: 'query', in: 'query', explode: true, schema: new OAT\Schema(ref: new Model(type: AccountFilter::class))

would generate working doc like this
![image](https://user-images.githubusercontent.com/11830741/167985190-02484ae3-bb9c-4830-bb9b-341e6e6034d9.png)

**Comment by PetrNovotnyGS** [comment]
Tried almost everything, but without success. My code is:
![image](https://user-images.githubusercontent.com/11830741/168643517-9fee55cd-f98a-4b80-8d1f-7a16b0ffdedb.png)

Result is:
![image](https://user-images.githubusercontent.com/11830741/168643328-b7b57bdc-e033-4375-8bea-1ff820a3df39.png)

**Comment by chrisguitarguy** [comment]
can you try `new OAT\Parameter(in: "query", name: "query", style: "form", explode: true, ref: new Model(....))`?

Basically remove the schema bit and put the `ref` directly into the parameter.

**Comment by PetrNovotnyGS** [comment]
new OAT\Parameter(name: 'query', in: 'query', ref: new Model(type: AccountFilterDTO::class), style: 'form', explode: true)
results in "mb_substr(): Argument #1 ($string) must be of type string, Nelmio\\ApiDocBundle\\Annotation\\Model given"

when used through alias
new OAT\Parameter(name: 'query', in: 'query', ref: '#/components/schemas/AccountFilter', style: 'form', explode: true)
doc is working, but parameters are empty.

**Comment by prugala** [comment]
@PetrNovotnyGS or someone else, do you have a solution maybe?

**Comment by simmstein** [comment]
Same issue. Have you a solution?

**Comment by adamgregory** [comment]
@GuilhemN @chrisguitarguy 
any chance we can reopen this issue because the issue in https://github.com/nelmio/NelmioApiDocBundle/issues/1721#issuecomment-1124488589 is not fixed?

**Comment by Luehrsen** [comment]
Resolving this is now possible with #2149 and has been released in `v4.14.0`. You'll need a custom processor to do that. This is roughly how you do it:

1) Generate a custom process as per the [official `swagger-php` examples](https://github.com/zircote/swagger-php/blob/master/Examples/processors/schema-query-parameter/SchemaQueryParameter.php) in your symfony app.
2) Register the processor with the tag `nelmio_api_doc.swagger.processor` [as described here](https://symfony.com/doc/current/service_container/tags.html).
3) The custom processor is now used in the generator.

Note: You can influence WHEN the processor is run by giving the tag a [`before`](https://github.com/zircote/swagger-php/blob/a70a5dc5db26183c86f589ff3d76fb11e141fc58/src/Generator.php#L306) attribute. The attribute takes the FQN of an existing processor and adds your custom processor right before it.
